sdk/cosmos/azure-cosmos-spark_3-0_2-12/src/main/scala/com/azure/cosmos/spark/AnalyzedFilter.scala

```diff
@@ -0,0 +1,17 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.cosmos.spark
+
+import com.azure.cosmos.models.CosmosParametrizedQuery
+import org.apache.spark.sql.sources.Filter
+
+// TODO: moderakh, thought for future optimization:
+//  if we can identify if the user filter is a equality on cosmos partitionKeyValue
+//  then we can set partitionKeyValue in the CosmosQueryOption
+//  the benefit is that if the partitionKeyValue is set in the CosmosQueryOption
+//  the antlr query parsing support can eliminate the need for query plan fetch from GW
```

Comment:
Same should be true for FeedRange query option (or in old model PkRangeId) as long as retry policies are in place, correct? Simply because we would never ever push down predicates that would need the QUery magic in the client - like Group By and Order BY - Spark is much better at doing them client-side :-)

Reply:
`AnalyzedFilter` analyzes "only" the user specified filters pushed down by spark platform. FeedRange/PkRangeId will not be a user push down query so hence won't be captured here. 

Yes, we need to capture add FeedRange/PkRangeId to the query options. That will happen elsewhere in the code where driver is submitting spark tasks:

need to be done here: 
https://github.com/Azure/azure-sdk-for-java/blob/feature/cosmos/spark30/sdk/cosmos/azure-cosmos-spark_3-0_2-12/src/main/scala/com/azure/cosmos/spark/CosmosScan.scala#L14-L28

Reply:
I think query plan fetch optimization only apply to logical partition key queries, since any other feed range still needs to go to gateway for the plan. The reason for this is because if you have a pkrangeid or epkrange, then you always have a chance to encounter a split, which makes our query go from one feed range to multiple. At this point you now need client side aggregation and without a query plan you can't pull it off.

Reply:
correct epkrange will be passed differently (as options) not through this code path.